spring-integration-file/src/main/java/org/springframework/integration/file/tail/OSDelegatingFileTailingMessageProducer.java

```diff
@@ -35,7 +36,7 @@
  *
  */
 public class OSDelegatingFileTailingMessageProducer extends FileTailingMessageProducerSupport
-		implements Runnable {
+		implements SchedulingAwareRunnable {
```

Comment:
If we are already here, would you mind to think if can avoid such a top level `Runnable` exposing?
I mean how about to move it to some inner class